### PR TITLE
Topic recognition - VQVAE OASIS brain dataset (Sam Suvichagorn s4699389)

### PR DESCRIPTION
**Pull request by ss-samiam**
This pull request works on the task of creating a generative model of the OASIS brain using a VQVAE that has a “reasonably clear image” and a Structured Similarity (SSIM) of over 0.6.

Results: My implementation has resulting SSIM of approximately 0.946 with an example of a decoded image as:
image

Files:
`images/` - contains images used in the README.md
`parameters.py` - hyperparameters for model and parameters for implementation
`dataset.py` - data preprocessing and loading into appropriate classes
`modules.py` - model architecture
`train.py` - trains the model with the dataset
`predict.py` - evaluates, validates, and give outputs of the trained model

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme:
- Project Overview: Good
- Model: Good
- Data + Preprocessing: Okay, but how was the data normalised? 
- Usage: Good
- Training / Loss Curve(s): Good
- Result Demonstration: Recon okay, but no image generation results
- References: Good

Functionality:
- Reconstruction: Good
- **Generation: Not completed**

Code:
- Consistent with Results: Yes
- File Structure: Good
- Commenting: Good
- Commit frequency: Good
- Commit messages: Good

Other Comments:
- Please restore recognition/Readme.MD which appears to have been deleted
- Indentation error in train.py at the start. Does the code run?

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem, no generation showed though -1
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage, no generation results -1
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe good
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good

**Comment by shakes76** [comment]
Merging needs the repo README restored, please fix. Does not affect grade only the merge.

**Comment by ss-samiam** [comment]
I have restored the repo README, as for feedback. Do I need to apply any? Or is there no further action required.

**Comment by wangzhaomxy** [comment]
Feedback attempt and no feedback marks lost.
